src/main/java/org/apache/mesos/executor/CustomExecutor.java

```diff
@@ -111,6 +115,45 @@ public void launchTask(final ExecutorDriver driver, final Protos.TaskInfo task)
         }
     }
 
+    private void executeHealthCheck(Protos.TaskInfo taskInfo, ExecutorTask executorTask) {
+        if (!taskInfo.hasHealthCheck()) {
+            LOGGER.info("No health check for task: " + taskInfo.getName());
+            return;
+        }
+
+        try {
+            HealthCheckMonitor healthCheckMonitor =
+                    new HealthCheckMonitor(
+                            HealthCheckHandler.create(
+                                    taskInfo,
+                                    scheduledExecutorService,
+                                    new HealthCheckStats(taskInfo.getName())),
+                            executorTask);
+            LOGGER.info("Submitting health check monitor.");
+            Future<Optional<HealthCheckStats>> futureOptionalHealthCheckStats =
+                    executorService.submit(healthCheckMonitor);
+
+            executorService.submit(new Runnable() {
```

Comment:
I moved the "exited without statistics" one.  Actually the thread here doesn't have much to do other than wait on the get() call.  Since for testing it's handing to be able to get() and receive statistics I don't want to remove that.  So since it's getting the stats it may as well log them here.  The Interrupted|ExecutionException case isn't avoidable due to the get().  Anyway, did what I could.